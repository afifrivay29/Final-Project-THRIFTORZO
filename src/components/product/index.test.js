import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Product from '../../components/product';

it('renders product card correctly', () => {
	const tree = renderer
		.create(
			<MemoryRouter>
				<Product
					img='http://res.cloudinary.com/amnesiac/image/upload/v1658150925/secondhand/product/ntuemunelbg9oedur6gr.jpg'
					title='Avanza'
					category='Kendaraan'
					price='250000000'
					link='5aace31a-3196-4d17-801f-017cd28ed7b1'
				/>
			</MemoryRouter>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
