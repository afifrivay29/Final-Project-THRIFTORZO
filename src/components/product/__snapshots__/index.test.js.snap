// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="/product/detail/5aace31a-3196-4d17-801f-017cd28ed7b1"
  onClick={[Function]}
>
  <div
    className="product-card pt-2 px-2 pb-4 rounded-[4px]"
  >
    <div
      className="product-card-image mb-2 relative"
    >
      <img
        alt="Avanza"
        className="rounded-[4px] product-image w-full h-28 object-cover"
        src="http://res.cloudinary.com/amnesiac/image/upload/v1658150925/secondhand/product/ntuemunelbg9oedur6gr.jpg"
      />
    </div>
    <h4
      className="text-sm text-black mb-1"
    >
      Avanza
    </h4>
    <span
      className="block mb-2 text-[10px] text-[#8A8A8A]"
    >
      Kendaraan
    </span>
    <span
      className="block text-sm text-black"
    >
      IDR 250,000,000.00
    </span>
  </div>
</a>
`;

exports[`renders product card correctly 1`] = `
<a
  href="/product/detail/5aace31a-3196-4d17-801f-017cd28ed7b1"
  onClick={[Function]}
>
  <div
    className="product-card pt-2 px-2 pb-4 rounded-[4px]"
  >
    <div
      className="product-card-image mb-2 relative"
    >
      <img
        alt="Avanza"
        className="rounded-[4px] product-image w-full h-28 object-cover"
        src="http://res.cloudinary.com/amnesiac/image/upload/v1658150925/secondhand/product/ntuemunelbg9oedur6gr.jpg"
      />
    </div>
    <h4
      className="text-sm text-black mb-1"
    >
      Avanza
    </h4>
    <span
      className="block mb-2 text-[10px] text-[#8A8A8A]"
    >
      Kendaraan
    </span>
    <span
      className="block text-sm text-black"
    >
      IDR 250,000,000.00
    </span>
  </div>
</a>
`;
